src/main/java/EventsAndCommands/FunEvents/GoodBotEvent.java

```diff
@@ -7,48 +7,50 @@
-                final String effectiveName = messageHistory.getRetrievedHistory().get(0).getMember().getEffectiveName();
-                final boolean isBot = messageHistory.getRetrievedHistory().get(0).getMember().getUser().isBot();
 
+    @Override
+    public void receiveCommand(GuildMessageReceivedEvent event) {
+        final TextChannel eventChannel = event.getChannel();
+        eventChannel.getHistoryBefore(event.getMessage(), 1).queue(messageHistory -> {
 
-                if (contentRaw.equalsIgnoreCase("good bot")) {
+            final String id = messageHistory.getRetrievedHistory().get(0).getMember().getUser().getId();
+            final String contentRaw = event.getMessage().getContentRaw();
+            final String effectiveName = messageHistory.getRetrievedHistory().get(0).getMember().getEffectiveName();
+            final boolean isBot = messageHistory.getRetrievedHistory().get(0).getMember().getUser().isBot();
 
-                    if (id.equalsIgnoreCase("503181069705805844")) {
+            if (contentRaw.equalsIgnoreCase("good bot")) {
 
-                        eventChannel.sendMessage("Thanks! :blush: :blush:  ").queue();
+                if (id.equalsIgnoreCase("503181069705805844")) {
 
-                    } else if (!isBot) {
+                    eventChannel.sendMessage("Thanks! :blush: :blush:  ").queue();
 
-                        eventChannel.sendMessage("I'm " + new Random().nextInt(101) + "% sure that "
-                                + effectiveName
-                                + " is not a bot").queue();
+                } else if (!isBot) {
 
-                    }
+                    eventChannel.sendMessage(
+                            "I'm " + new Random().nextInt(101) + "% sure that " + effectiveName + " is not a bot")
+                            .queue();
 
-                } else if (id.equalsIgnoreCase("503181069705805844") && contentRaw.equalsIgnoreCase("bad bot")) {
-                    eventChannel.sendMessage("No u").queue();
-                } else if (!isBot && contentRaw.equalsIgnoreCase("bad bot")) {
-                    eventChannel.sendMessage("I'm " + new Random().nextInt(101) + "% sure that " + effectiveName + " is not a bot").queue();
                 }
 
-            });
-
-
-
-        }
+            } else if (id.equalsIgnoreCase("503181069705805844") && contentRaw.equalsIgnoreCase("bad bot")) {
+                eventChannel.sendMessage("No u").queue();
+            } else if (!isBot && contentRaw.equalsIgnoreCase("bad bot")) {
+                eventChannel.sendMessage(
+                        "I'm " + new Random().nextInt(101) + "% sure that " + effectiveName + " is not a bot")
+                        .queue();
+            }
 
+        });
+    }
 
+    @Override
+    public boolean trigger(GuildMessageReceivedEvent event) {
+        String msg = event.getMessage().getContentRaw().toLowerCase();
+        boolean r = (msg.contains("good bot") || msg.contains("bad bot"));
```

Comment:
You don't need the local variable here, you can just return the boolean expression.